### PR TITLE
Fixes ambient sound preferences runtiming every time a client connects

### LINKED ISSUE #84200: ship ambience preference runtimes every time a client connects.
image

#84087 @MrMelbert 

```
[12:25:42] Runtime in code/modules/client/client_procs.dm, line 1132: Cannot execute null.read preference().
proc name: update ambience pref (/client/proc/update_ambience_pref)
src: Shmesser (/client)
call stack:
Shmesser (/client): update ambience pref()
/datum/preference/toggle/sound... (/datum/preference/toggle/sound_ambience): apply to client(Shmesser (/client), 1)
/datum/preferences (/datum/preferences): apply all client preferences()
/datum/preferences (/datum/preferences): load preferences()
/datum/preferences (/datum/preferences): New(Shmesser (/client))
Shmesser (/client): New(null)
```

### PR DESCRIPTION
**Pull request by ShizCalev**
fixes #84200

forgot to pass the value that's sent while preferences are still initializing.

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -1128,8 +1128,8 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 		winset(src, "default.Shift", "is-disabled=true")
 		winset(src, "default.ShiftUp", "is-disabled=true")
 
-/client/proc/update_ambience_pref()
-	if(prefs.read_preference(/datum/preference/toggle/sound_ambience))
+/client/proc/update_ambience_pref(value)
+	if(value || prefs.read_preference(/datum/preference/toggle/sound_ambience))
```
Comment (MrMelbert) [inline code/modules/client/client_procs.dm:1132]: First of all - wtf `apply_to_client` is called before the mob even gets prefs???

Second of all - if `value` is `FALSE`, won't this still runtime?

Comment (MrMelbert) [reply code/modules/client/client_procs.dm:1132]: Maybe we just use `apply_to_client_updated`

Comment (ShizCalev) [reply code/modules/client/client_procs.dm:1132]: > First of all - wtf `apply_to_client` is called before the mob even gets prefs???
apply_all_client_preferences() is what's SENDING the value of the preference during initial client/new, later calls can properly use prefs.red_preference
https://github.com/tgstation/tgstation/blob/2f2b74338cb7397e4128ccdef00bbc4d6c67a0a5/code/modules/client/client_procs.dm#L242-L273
https://github.com/tgstation/tgstation/blob/2f2b74338cb7397e4128ccdef00bbc4d6c67a0a5/code/modules/client/preferences.dm#L329-L337
> 
> Second of all - if `value` is `FALSE`, IE they have the pref toggled off, won't this still runtime?
Only the very first call of a player's savefile is the issue here, after that the preference is set.



>Maybe we just use apply_to_client_updated()
image
same issue.

Comment (ShizCalev) [reply code/modules/client/client_procs.dm:1132]: infact one sec, can just eliminate this check completely.

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**